### DISCUSSION TITLE
Is there any way to convert PDF into picture using QuestPDF?

### DISCUSSION THREAD

**Discussion by GirlWithAnEpee** [QuestPDF/QuestPDF#1048] [Q&A] [answered]
Here https://www.questpdf.com/concepts/generating-output.html#generating-images there's way to generate a PDF-file and then save it as PDF, image, etc. But what if the file is not generated with QuestPDF but uploaded for example by a user, how to "ask" QuestPDF to convert it?

**Comment by DawidRobertNowak** [comment, marked as answer]
Hello,

We are closing this thread due to inactivity. We apologize for the delayed response. Thank you for taking the time to reach out and contribute to the project — your feedback helps us improve QuestPDF.

The `GenerateImages()` method only works with documents generated by QuestPDF. For converting existing PDF files to images, a dedicated PDF rendering library would be needed, as this is outside QuestPDF's scope. We do not have plans to introduce rendering existing PDF files as images.

Should you encounter any further problems, please don't hesitate to open a new issue.

Thank you!
